docs/advanced-concepts/route-meta.md

```diff
@@ -1 +1,42 @@
 # Route Meta
+
+It may be useful to store additional context on your route to be used in a hook, or within a component. Meta data might be useful for authorization, analytics, and much more. For example, below we will use route meta to configure the document title per route.
+
+```ts
+import { createRoutes, createRouter } from '@kitbag/router'
+
+const routes = createRoutes([
+  { 
+    name: 'home',
+    path: '/',
+    component: Home,
+    meta: {
+      pageTitle: 'Kitbag Home'
+    }
+  },
+  { 
+    name: 'path',
+    path: '/about',
+    component: About,
+    meta: {
+      pageTitle: 'Learn More About Kitbag'
+    }
+  },
+])
+
+const router = createRouter(routes)
+
+router.onAfterRouteEnter(to => {
+  document.title = to.matched.meta.pageTitle
+})
+```
+
+You might notice that the default type for `pageTitle` is `unknown`. By default meta is `Record<string, unknown>` but can be narrowed with [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html).
+
+```ts
+declare module '@kitbag/router' {
+  interface RouteMeta {
+    pageTitle?: string
+  }
```

Comment:
In my head I thought this was part of the Register interface for some reason. Prob makes sense that it's not since it's part of that interface since meta is part of routes which is used to create the router.

Reply:
same. I considered combining it but ultimately decided not to. Do you feel strongly?

Reply:
I'm honestly not even sure it would work. If it does, it kinda be nice to be consistent and have everything go through Register. But I don't feel strongly either way. The devx and outcome is about the same